src/phpDocumentor/Transformer/Template.php

```diff
@@ -129,7 +129,20 @@ public function setVersion(string $version) : void
         $this->version = $version;
     }
 
-    public function files() : FilesystemInterface
+    /**
+     * FlySystem filesystem / MountManager containing the template files, base templates files
+     * and destination filesystem.
+     *
+     * This MountManager has three mounts:
+     *
+     * - template://, the files of this template
+     * - templates://, the base folder containing phpDocumentor's global templates (i.e. `/data/templates`)
```

Comment:
I think this is because you need some files from the global templates to be used during the rendering. It would have been nice if we could have this more like a transparent file system. Because you might know which directory you need to use to get the assets you need? So if `template://` doesn't contain the file you need look in `templates://`. And hide this from the actual implementation? So writers only need `template://`
If you see any options to do this that would be nice. but it is also ok to keep it like this for now.

Reply:
Not entirely; the difference between these two is the difference between `/data/templates` and `/data/templates/clean`. When a path starts with `templates/` it actually refers to the 'global templates path'; where if `templates/` is not in front of it, it is considered to be the template folder itself. So there is no one-on-one overloading happening here.